### ISSUE TITLE
[thumb:] is not supported on guide pages

### ISSUE THREAD

**Issue by kleintom** [bugguide/bugguide#438]
If I locally add a [thumb:] to the Identification section of a guide page, for example, it shows up on the page literally as '[thumb:]', with no added markup. Presumably that's because [thumb:] processing has been implemented in the bgimage module: https://github.com/bugguide/bugguide/blob/58cf8a056e0fe4b9ab348bf510557b104c926884/sites/all/modules/custom/bgimage/bgimage.module#L595-L610 but not in the bgpage module.

For migrated guide pages, I see slightly different behavior: the thumbs have been turned into links, but the link address is incomplete (as far as I know), so the thumbs still don't appear. Compare https://beta.bugguide.net/node/342391 with https://bugguide.net/node/view/342391 for example - here's a [thumb:] link on beta, note the 'http:' instead of 'https:' and a '.' in place of the expected 'beta.bugguide.net'
![image](https://user-images.githubusercontent.com/632915/146648328-22538b18-95b4-44cc-8452-f7722ffec044.png)
(if you change the '.' to beta.bugguide.net in the inspector panel, the thumb appears).

**Comment by kleintom** [comment]
Coincidentally I just noticed that firefox (but not chrome) shows a warning that 'parts of this page are not secure (such as images)':
![image](https://user-images.githubusercontent.com/632915/146648594-e3a8c6fb-8e9a-4998-b193-8d833d36a88d.png)
Presumably that's because of the http:// thumb images.

**Comment by kleintom** [comment]
> If I locally add a [thumb:] to the Identification section of a guide page, for example, it shows up on the page literally as '[thumb:]', with no added markup. Presumably that's because [thumb:] processing has been implemented in the bgimage module but not in the bgpage module.

Nope, that's not how that works. From the hook_filter_info page: "User submitted content is passed through a group of filters before it is output in HTML, in order to remove insecure or unwanted parts, correct or enhance the formatting, transform special keywords, etc. A group of filters is referred to as a "text format". Administrators can create as many text formats as needed. Individual filters can be enabled and configured differently for each text format."

You can see available text formats at admin/config/content/formats - if you configure an existing text format or create a new one, you see that all filters announced by any module's hook_filter_info are available to add to any text format.

The format we use for formatted text fields is named 'f' for 'filtered html' [if I recall correctly, it's one letter to reduce database size, since the format is listed with every field entry]: admin/config/content/formats/f It has the thumb filter applied:
![image](https://user-images.githubusercontent.com/632915/146657354-f368d4ff-2f66-4fca-89b7-0aeab052bda7.png)

The text format of a specific field is set in the settings for that field, for example the text format for the Identification field of a guide page is set at admin/structure/types/manage/bgpage/fields/field_id
![image](https://user-images.githubusercontent.com/632915/146657493-5226fe1d-2ca1-40ab-bbe4-f2652ee2c8cd.png)
Many(?) of our text formats are not set explicitly (by setting 'Limit allowed text formats' to a single format); rather they specify that we should use _some_ text format, which would normally mean the user gets to choose the text format they would like to use from a dropdown that appears below the text entry box, but in our case the only text format that's allowed to be used by non-admins is the 'f' format:
![image](https://user-images.githubusercontent.com/632915/146657546-c631fc36-627f-428b-ad71-b1c104dfe005.png)
so no text format selection box appears if you're not admin - you just automatically get 'f'.

**Comment by kleintom** [comment]
The reason I just saw the unformatted text '[thumb:]' when I tried to add a thumb to a guide page locally was that I tried to do it as admin, and the text format that was auto-selected for the Identification field (without me even noticing) was xbbcode, which does not apply the thumb filter. Once I changed the format to 'f' it worked as expected for admin, and it worked automatically as expected when I logged in as editor.

And now I see thumbs are back at https://beta.bugguide.net/node/342391 :-) Well good reminder for me of how the filter/text format system works!
